src/VirtoCommerce.Platform.Data/ExportImport/JsonSerializerExtensions.cs

```diff
@@ -52,9 +52,8 @@ public static async Task DeserializeJsonArrayWithPagingAsync<T>(this JsonTextRea
                 {
                     cancellationToken.ThrowIfCancellationRequested();
 
-                    var itemType = AbstractTypeFactory<T>.TryCreateInstance().GetType();
-                    var item = serializer.Deserialize(reader, itemType);
-                    items.Add((T)item);
+                    var item = serializer.Deserialize<T>(reader);
```

Comment:
This change breaks runtime types overriding via AbstractTypefactory since all objects will be instantiated with `T` based type and won't respect runtime overrides.

example:

Module-A
 void MyCoolMethod(A  param);

Module-B
`B` derived from `A` overrides via AbstractTypeFactory<A>.Override<A, B>()

Result (runtime)
MyCoolMethod receives `param` with type `A` instead of `B` as expected.

Reply:
Please ignore this comment, I realized that I was mistaken

Reply:
You freaked me out.